Branch-SDK/Branch-SDK/BNCServerRequest.m

```diff
@@ -1,61 +1,29 @@
 //
 //  BNCServerRequest.m
-//  Branch-SDK
+//  Branch-TestBed
 //
-//  Created by Alex Austin on 6/5/14.
-//  Copyright (c) 2014 Branch Metrics. All rights reserved.
+//  Created by Graham Mueller on 5/22/15.
+//  Copyright (c) 2015 Branch Metrics. All rights reserved.
 //
 
 #import "BNCServerRequest.h"
-#import "BNCPreferenceHelper.h"
-
-#define TAG         @"TAG"
-#define DATA        @"POSTDATA"
-
-@interface BNCServerRequest() <NSCoding>
-
-@end
-
 
 @implementation BNCServerRequest
 
-- (void)encodeWithCoder:(NSCoder *)coder {
-    if (self.tag) {
-        [coder encodeObject:self.tag forKey:TAG];
-    }
-    if (self.postData) {
-        [coder encodeObject:self.postData forKey:DATA];
-    }
-}
-
-- (id)initWithCoder:(NSCoder *)coder {
-    if (self = [super init]) {
-        self.tag = [coder decodeObjectForKey:TAG];
-        self.postData = [coder decodeObjectForKey:DATA];
-    }
-    return self;
+- (void)makeRequest:(BNCServerInterface *)serverInterface key:(NSString *)key callback:(BNCServerCallback)callback {
+    [NSException raise:@"Improper BNCServerRequest" format:@"BNCServerRequest subclasses must implement makeRequest:key:callback:"];
 }
 
-- (id)initWithTag:(NSString *)tag {
-    return [self initWithTag:tag andData:nil];
+- (void)processResponse:(BNCServerResponse *)response error:(NSError *)error {
```

Comment:
The way I see it, the request knows how to construct itself and send to the server, and how to process the response. That way, as a consumer all I do is this:

```
MyRequest *req = [[MyRequest alloc] initWithCallback:^(NSDictionary *params, NSError *error) {
    if (err) {
        NSLog(@"Request Failed");
    }
    else {
        NSLog("Request completed with params: %@", params);
    }
}];

[req send];// (or in our case, enqueue)
```